tla/consensus/ccfraft.tla

```diff
@@ -941,7 +941,7 @@ RejectAppendEntriesRequest(i, j, m, logOk) ==
                              dest           |-> j],
                              m)
                    \/ /\ prevTerm # 0
-                      /\ LET lli == FindHighestPossibleMatch(log[i], m.prevLogIndex, m.term)
+                      /\ LET lli == FindHighestPossibleMatch(log[i], m.prevLogIndex, m.prevLogTerm)
```

Comment:
This was the source of `suffix_collision.3` failing trace validation. The AE claims that `{m.prevLogTerm}.{m.prevLogIdx}` is a TxID in its history. If we NACK, we want to find the highest entry we hold which could agree with that. The previous statement was trying to find a match with `{m.term}.{m.prevLogIndex}`, which is an invented TxID. In most instances it was also valid, or even matched, but we luckily have one case where it doesn't.

Reply:
This sounds significant enough to, perhaps, go into a dedicated/separate PR.